request-server.go

```diff
@@ -199,6 +199,12 @@ func cleanPacketPath(pkt *sshFxpRealpathPacket) responsePacket {
 
 func cleanPath(path string) string {
```

Comment:
`cleanPath` is called for more than just SSH_FXP_REALPATH. In fact, it looks like commit 4d7bb970c49d8a86d12e2cd34952dc80dbc7cb0d made a logical mistake when they removed the duplicate codes of `filepath.ToSlash(filepath.Clean(path))` in favor of `cleanPath(path)`, as the behavior of those calls was explicitly _not_ supposed to append a '/' to the front of the paths. (Only RequestServer.SSH_FXP_REALPATH was supposed to do this.)

I am concerned that this code would just further exacerbate the problem rather than provide a root-cause fix. When `cleanPath` was only working on an FXP_REALPATH call, its assumptions on constructing an absolute path were “acceptable”, but then the code got pushed around and used elsewhere, where the expanded behavior of turning relative paths into absolute paths did not exist before.

Properly, all of the FXP_REALPATH code paths should be calling cleanPath only _after_ it turns it from relative to absolute, and this functionality should not be done itself in the cleanPath function at all. (Even if it looks like in the RequestServer this amounts simply to “prepend a slash to the path".)